### PR TITLE
docs: migration guide to the `latest` [1/n]

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

File: docs/source-pytorch/upgrade/1_8.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
+:orphan:
+
+Upgrade from 1.8 to the latest
+##############################
+
+Regular User
+************
+
+.. include:: xuser/regular_1_8.rst
```
Comment (Borda) [inline docs/source-pytorch/upgrade/1_8.rst:9]: example of aggregation...

File: docs/source-pytorch/upgrade/sections/1_4_devel.rst
Comment (justusschock) [inline docs/source-pytorch/upgrade/sections/1_4_devel.rst:1]: @Borda why is this empty?

File: docs/source-pytorch/index.rst [hunk 2]
```diff
@@ -181,6 +182,7 @@ Current Lightning Users
    :caption: API Reference
 
    api_references
+   upgrade/migration_guide
```
Comment (Borda) [inline docs/source-pytorch/index.rst:185]:
```suggestion
```

File: docs/source-pytorch/upgrade/sections/1_4_regular.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+   :widths: 40 40 20
+   :header-rows: 1
+
+   * - If
+     - Then
+     - Ref
+
+   * - relied on the ``outputs`` in your  ``LightningModule.on_train_epoch_end`` or ``Callback.on_train_epoch_end`` hooks
+     - rely on either ``on_training_epoch_end`` or set outputs as attributes in your ``LightningModule`` instances and access them from the hook
+     - #7339
```
Comment (justusschock) [inline docs/source-pytorch/upgrade/sections/1_4_regular.rst:11]: as follow-up: can we add links to these?

**Review by justusschock — APPROVED** [review]

File: docs/source-pytorch/past_versions.rst [hunk 4]
```diff
@@ -3,21 +3,24 @@ Past PyTorch Lightning versions
...
 
 .. list-table:: Past versions
-   :widths: 10 50 40
+   :widths: 5 50 30 15
    :header-rows: 1
 
    * - Version
      - Title
      - Bug-fix versions
+     - upgrade guide
```
Comment (carmocca) [inline docs/source-pytorch/past_versions.rst:15]:
```suggestion
     - Upgrade guide
```

File: docs/source-pytorch/upgrade/migration_guide.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
+Guide how to upgrade to the 2.0 version
+#######################################
+
+IN the following section we will guide use how to update from almost any Lightning version to the 2.0 release.
```
Comment (carmocca) [inline docs/source-pytorch/upgrade/migration_guide.rst:4]:
```suggestion
The following section will guide you through updating to the 2.0 release.
```

**Review by carmocca — APPROVED** [review]
No time to review the items. Structure looks good

File: docs/source-pytorch/upgrade/sections/1_4_regular.rst (hunk 3 above)
Comment (Borda) [reply docs/source-pytorch/upgrade/sections/1_4_regular.rst:11]: yes

File: docs/source-pytorch/upgrade/sections/1_4_devel.rst
Comment (Borda) [reply docs/source-pytorch/upgrade/sections/1_4_devel.rst:1]: there were no changes for this level, we may regoup

**Pull request merged**